src/app/shared/helpers/data-pager.ts

```diff
@@ -21,19 +21,26 @@ interface PagerOptions<T> {
   onLoadMoreError: (error: Error) => void,
 }
 
+interface PagedData<T> {
+  list: T[],
+  newItems: T[],
+}
+
 export class DataPager<T> {
   private trigger$ = new ReplaySubject<{ reset: boolean }>(1);
 
-  state$ : Observable<FetchState<T[]>> = this.trigger$.pipe(
-    switchScan<{ reset: boolean }, FetchState<T[]>, Observable<FetchState<T[]>>>(
+  private state$ : Observable<FetchState<PagedData<T>>> = this.trigger$.pipe(
+    switchScan<{ reset: boolean }, FetchState<PagedData<T>>, Observable<FetchState<PagedData<T>>>>(
       (prev, { reset }) => {
         prev = reset ? fetchingState() : prev;
-        const latestElement = prev.data !== undefined ? prev.data[prev.data.length-1] : undefined;
-        return this.options.fetch(latestElement).pipe(
+        const latestElement = prev.data?.list[prev.data.list.length-1];
+        return this.options.fetch(this.options.pageSize, latestElement).pipe(
           mapToFetchState(),
-          map<FetchState<T[]>, FetchState<T[]>>(state => {
-            // Case 1: First fetch -> return state as is
-            if (prev.data === undefined) return state;
+          map((state): FetchState<PagedData<T>> => {
+            // Case 1: First fetch
+            if (prev.data === undefined) {
+              return state.isReady ? readyState({ list: state.data, newItems: state.data }) : state as FetchState<PagedData<T>>;
```

Comment:
Use of `as` is a bad smell, a mistake most of time... let's check this...
If VSCode is correct, `state` is a `FetchState<T[]>`, so considering it as a `FetchState<PagedData<T>>` seems incorrect.

Reply:
fixed